src/OrbitBase/StopSourceTest.cpp

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest-death-test.h>
+#include <gtest/gtest.h>
+
+#include "OrbitBase/StopSource.h"
+#include "OrbitBase/StopToken.h"
+
+namespace orbit_base {
+
+TEST(StopSource, ConstructStopPossible) {
+  StopSource stop_source{};
+  EXPECT_TRUE(stop_source.IsStopPossible());
+}
+
+TEST(StopSource, CopyStopPossible) {
+  {
+    StopSource stop_source{};
+    StopSource stop_source_copy{stop_source};
+    EXPECT_TRUE(stop_source.IsStopPossible());
+    EXPECT_TRUE(stop_source_copy.IsStopPossible());
+  }
+  {
+    StopSource stop_source{};
+    StopSource stop_source_copy = stop_source;
+    EXPECT_TRUE(stop_source.IsStopPossible());
+    EXPECT_TRUE(stop_source_copy.IsStopPossible());
+  }
+}
+
+TEST(StopSource, MoveStopPossible) {
+  {
+    StopSource stop_source{};
+    StopSource stop_source_moved{std::move(stop_source)};
+    EXPECT_FALSE(stop_source.IsStopPossible());
+    EXPECT_TRUE(stop_source_moved.IsStopPossible());
+  }
+  {
+    StopSource stop_source{};
+    StopSource stop_source_moved = std::move(stop_source);
+    EXPECT_FALSE(stop_source.IsStopPossible());
+    EXPECT_TRUE(stop_source_moved.IsStopPossible());
+  }
+}
+
+TEST(StopSource, InvalidAccessAfterMove) {
+  StopSource stop_source;
+  StopSource stop_source_moved{std::move(stop_source)};
+
+  EXPECT_DEATH(stop_source.RequestStop(), "");
+  EXPECT_DEATH((void)stop_source.GetStopToken(), "");
+}
```

Comment:
I wouldn't test this as this is usually a behaviour we don't want to guarantee. By having this test it basically means we guarantee that we crash and that this is part of the public facing behaviour users can and should rely on.

Reply:
removed it